backend/src/main/java/ai/verta/modeldb/batchProcess/DatasetToRepositoryMigration.java

```diff
@@ -0,0 +1,270 @@
+            } catch (Exception e) {
+              e.printStackTrace();
+              LOGGER.error(e.getMessage());
+            }
+          }
+        } else {
+          LOGGER.debug("Datasets total count 0");
+        }
+
+        transaction.commit();
+        lowerBound += pagesize;
+      } catch (Exception ex) {
+        if (ModelDBUtils.needToRetry(ex)) {
+          migrateDatasetsToRepositories();
+        } else {
+          throw ex;
+        }
+      }
+    }
+
+    LOGGER.debug("Datasets To Repositories migration finished");
+  }
+
+  private static void createRepository(
+      Session session, DatasetEntity datasetEntity, UserInfo userInfoValue)
+      throws ModelDBException, NoSuchAlgorithmException, InvalidProtocolBufferException {
+    String datasetId = datasetEntity.getId();
+    Dataset newDataset = datasetEntity.getProtoObject().toBuilder().setId("").build();
+    Repository repository;
+    try {
+      repository =
+          repositoryDAO.createRepository(commitDAO, metadataDAO, newDataset, true, userInfoValue);
+    } catch (Exception e) {
+      if (e instanceof StatusRuntimeException) {
+        Status status = Status.fromThrowable(e);
+        if (status.getCode().equals(Status.Code.ALREADY_EXISTS)) {
+          repository =
+              repositoryDAO.createRepository(commitDAO, metadataDAO, newDataset, false, null);
+        } else {
+          throw e;
+        }
+      } else {
+        throw e;
+      }
+    }
+    LOGGER.debug("Migration done for dataset to repository - Repo ID : {}", repository.getId());
+    migrateDatasetVersionToCommitsBlobsMigration(session, datasetId, repository.getId());
+  }
+
+  private static void migrateDatasetVersionToCommitsBlobsMigration(
+      Session session, String datasetId, Long repoId) {
+    LOGGER.info("DatasetVersions To Commits and Blobs migration started");
+    String countQuery =
+        "SELECT COUNT(dv) FROM "
+            + DatasetVersionEntity.class.getSimpleName()
+            + " dv WHERE dv.dataset_id = :datasetId";
+    Query query = session.createQuery(countQuery);
+    query.setParameter("datasetId", datasetId);
+    Long count = (Long) query.uniqueResult();
+
+    int lowerBound = 0;
+    final int pagesize = recordUpdateLimit;
+    LOGGER.debug("Total DatasetVersions {}", count);
+
+    while (lowerBound < count) {
+
+      try (Session session1 = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+        Transaction transaction = session1.beginTransaction();
+        CriteriaBuilder criteriaBuilder = session1.getCriteriaBuilder();
+
+        CriteriaQuery<DatasetVersionEntity> criteriaQuery =
+            criteriaBuilder.createQuery(DatasetVersionEntity.class);
+        Root<DatasetVersionEntity> root = criteriaQuery.from(DatasetVersionEntity.class);
+
+        CriteriaQuery<DatasetVersionEntity> selectQuery =
+            criteriaQuery
+                .select(root)
+                .where(criteriaBuilder.equal(root.get("deleted"), false))
```

Comment:
same as dataset let us not delete the ole entities

Reply:
Yes I have already removed that code so currently, we are not deleting old entities now